src/components/Question/index.tsx

```diff
@@ -7,50 +7,46 @@ import Option from './Option';
 import styles from './styles.module.css';
 
 interface Props {
-  text: string;
-  options: IOption[];
   submitResponse: Function;
-  isAttempted: boolean;
-  isCorrect: boolean;
-  correctOptionHash?: string;
   onClose: Function;
-  preSelectedChoice?: string;
   isPreview?: boolean;
   isQuestionSaved?: boolean;
-  isWithoutOptions?: boolean;
   pauseTimer?: Function;
+  selectedOptionId: number | null | undefined | string;
+  selectedQuestion: {
+    questionId?: string;
+    text: string;
+    options: IOption[];
+  };
+  isWithoutOptions: boolean;
+  isQuestionTimerUp?: boolean;
+  isAttempted: boolean;
 }
 
 export default function Question({
-  text,
-  options,
   submitResponse,
-  isAttempted,
-  isCorrect,
-  correctOptionHash = '',
   onClose,
-  preSelectedChoice = '',
   isPreview = false,
   isQuestionSaved = true,
-  isWithoutOptions = false,
   pauseTimer,
+  selectedOptionId,
+  selectedQuestion,
+  isWithoutOptions,
+  isQuestionTimerUp,
+  isAttempted,
 }: Props) {
-  const [selectedChoice, setSelectedChoice] = useState(preSelectedChoice);
-  const [isAnswerRevealed, setIsAnswerRevealed] = useState(isPreview);
-
-  useEffect(() => {
-    // sync selected choice from props with state when changed from the edit component
-    setSelectedChoice(preSelectedChoice);
-  }, [preSelectedChoice]);
+  const [selectedChoice, setSelectedChoice] = useState(null);
+  const [isAnswerRevealed, setIsAnswerRevealed] = useState(false);
+  const { options, questionId, text } = selectedQuestion;
 
   useEffect(() => {
-    if (isAttempted && !isPreview) {
+    if (!selectedOptionId) {
       // reveal answer when question is attempted due to timeout
-      setIsAnswerRevealed(true);
+      setIsAnswerRevealed(!!isQuestionTimerUp);
```

Comment:
When timer is up then we were setting isAttempted. Are we not doing that anymore?

Reply:
We should just keep the earlier code. No change required here.

Reply:
reverted